samtranslator/validator/resource_validator.py

```diff
@@ -0,0 +1,40 @@
+"""A resource validator to help validate resource properties and raise exception when some value is unexpected."""
+from typing import Any, Dict, Type
+
+from pydantic import BaseModel
+
+
+class ResourceModel:
+    """
+    Wrapper class around a SAM schema BaseModel to with a new functional "get" method
+    """
+
+    def __init__(self, model: BaseModel) -> None:
+        self.model = model
+
+    def get(self, attr: str, default_value: Any = None) -> Any:
```

Comment:
This isn't a great DX as there's no autocompletion I assume. Any way to improve?

Reply:
I don't think we need to handle autocompletion in this PR. We can have followup PR to make it more user friendly.